File: GPE_problems/gpe_23681_Bachet's Game/gpe_23681.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n, m;
    while (cin >> n >> m) {
        vector<int> moves(m);
        for (int i = 0; i < m; ++i) cin >> moves[i];

        vector<bool> dp(n + 1, false);

        for (int i = 1; i <= n; ++i) {
            for (int move : moves) {
                if (i - move >= 0 && !dp[i - move]) {
                    dp[i] = true;
                    break;
                }
            }
        }

        if (dp[n]) cout << "Stan wins\n";
        else cout << "Ollie wins\n";
    }
    return 0;
}

File: GPE_problems/gpe_23681_Bachet's Game/gpe_23681.py
import sys

for line in sys.stdin:
    if not line.strip():
        continue

    token = list(map(int,line.strip().split()))
    n, m = token[0], token[1]
    valid_move = token[2:]

    dp = [False] * (n+1)

    
    for i in range(1, n+1):
        for move in valid_move:
            if i - move and dp[i-move] == False:
                dp[i] = True
                break

    if dp[n] == True:
        print("Stan wins")
    else:
        print("Ollie wins")